jenkins/dockerized-e2e-runner.sh

```diff
@@ -60,8 +60,16 @@ if [[ "${KUBE_RUN_FROM_OUTPUT:-}" =~ ^[yY]$ ]]; then
     )
 fi
 
+# Timeouts can leak the container, causing weird issues where the job keeps
+# running and *deletes resources* that the next job is using.
+# Give the container a unique name, and make bash kill it on EXIT,
+# which will happen during a timeout.
+CONTAINER_NAME="${JOB_NAME}-${BUILD_NUMBER}"
+trap "docker kill ${CONTAINER_NAME}" EXIT
```

Comment:
[Yes.](https://github.com/kubernetes/test-infra/blob/master/jenkins/e2e-image/e2e-runner.sh#L360-L363)

Note that our outer kill timeout is only 15s.